sklearn/feature_selection/from_model.py

```diff
@@ -200,7 +230,7 @@ def _get_support_mask(self):
                 ' while passing the fitted estimator to the constructor.')
         scores = _get_feature_importances(estimator)
         self.threshold_ = _calculate_threshold(estimator, scores,
-                                               self.threshold)
+                                               self.threshold, self.k)
```

Comment:
As @jnothman says, this might lead to more than `k` values being selected if there is a tie for feature importances. It might be better to compute the mask here using an argsort independently of `_calculate_threshold`.